### PR TITLE
filterByContentNode move Joins to where-clause

### LINKED ISSUE #4931: slow query: ContentNode
```sql

SELECT c0_.id            AS id_0,
       c0_.createTime    AS createtime_1,
       c0_.updateTime    AS updatetime_2,
       c0_.data          AS data_3,
       c0_.slot          AS slot_4,
       c0_.position      AS position_5,
       c0_.instanceName  AS instancename_6,
       c1_.id            AS id_7,
       c1_.createTime    AS createtime_8,
       c1_.updateTime    AS updatetime_9,
       c1_.data          AS data_10,
       c1_.slot          AS slot_11,
       c1_.position      AS position_12,
       c1_.instanceName  AS instancename_13,
       c2_.id            AS id_14,
       c2_.createTime    AS createtime_15,
       c2_.updateTime    AS updatetime_16,
       c2_.data          AS data_17,
       c2_.slot          AS slot_18,
       c2_.position      AS position_19,
       c2_.instanceName  AS instancename_20,
       c3_.id            AS id_21,
       c3_.createTime    AS createtime_22,
       c3_.updateTime    AS updatetime_23,
       c3_.data          AS data_24,
       c3_.slot          AS slot_25,
       c3_.position      AS position_26,
       c3_.instanceName  AS instancename_27,
       c4_.id            AS id_28,
       c4_.createTime    AS createtime_29,
       c4_.updateTime    AS updatetime_30,
       c4_.name          AS name_31,
       c4_.active        AS active_32,
       c4_.entityClass   AS entityclass_33,
       c4_.jsonConfig    AS jsonconfig_34,
       c0_.strategy      AS strategy_35,
       c0_.rootId        AS rootid_36,
       c0_.parentId      AS parentid_37,
       c0_.contentTypeId AS contenttypeid_38,
       c1_.strategy      AS strategy_39,
       c1_.rootId        AS rootid_40,
       c1_.parentId      AS parentid_41,
       c1_.contentTypeId AS contenttypeid_42,
       c2_.strategy      AS strategy_43,
       c2_.rootId        AS rootid_44,
       c2_.parentId      AS parentid_45,
       c2_.contentTypeId AS contenttypeid_46,
       c3_.strategy      AS strategy_47,
       c3_.rootId        AS rootid_48,
       c3_.parentId      AS parentid_49,
       c3_.contentTypeId AS contenttypeid_50
FROM content_node c0_
         INNER JOIN content_node c1_ ON c0_.rootId = c1_.id AND c1_.strategy IN ($5)
         INNER JOIN activity a5_ ON (a5_.rootContentNodeId = c1_.id)
         INNER JOIN schedule_entry s6_ ON a5_.id = s6_.activityId AND (s6_.periodId = $1)
         LEFT JOIN content_node c2_ ON c0_.parentId = c2_.id AND c2_.strategy IN ($6, $7, $8, $9, $10, $11, $12)
         LEFT JOIN content_node c3_ ON c0_.id = c3_.parentId AND c3_.strategy IN ($13, $14, $15, $16, $17, $18, $19)
         INNER JOIN content_type c4_ ON c0_.contentTypeId = c4_.id
         INNER JOIN content_node c7_ ON c0_.rootId = c7_.id AND c7_.strategy IN ($20)
         LEFT JOIN activity a8_ ON (a8_.rootContentNodeId = c7_.id)
         INNER JOIN category c9_ ON (c9_.id = a8_.categoryId OR c9_.rootContentNodeId = c7_.id)
         INNER JOIN camp c10_ ON c9_.campId = c10_.id
         LEFT JOIN camp_collaboration c11_ ON c10_.id = c11_.campId AND (c11_.userId = $2 AND c11_.status = $3)
WHERE (c11_.id IS NOT NULL OR c10_.isPrototype = $4)
  AND c0_.strategy IN ($21, $22, $23, $24, $25, $26, $27)
ORDER BY c0_.createTime DESC, c1_.id ASC, c2_.id ASC, c0_.slot ASC, c0_.position ASC;
```

### PR DESCRIPTION
**Pull request by pmattmann**
fixes #4931

### CODE REVIEW

File: api/migrations/schema/Version20240413142253.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
```
Comment (BacLuc) [inline api/migrations/schema/Version20240413142253.php]: You could also delete the generated comments.

File: api/src/Entity/Camp.php [hunk 2]
```diff
@@ -125,6 +125,9 @@ class Camp extends BaseEntity implements BelongsToCampInterface, CopyFromPrototy
     #[ORM\OneToMany(targetEntity: MaterialList::class, mappedBy: 'camp', orphanRemoval: true, cascade: ['persist'])]
     public Collection $materialLists;
 
+    #[ORM\OneToMany(targetEntity: CampRootContentNode::class, mappedBy: 'camp')]
```
Comment (BacLuc) [inline api/src/Entity/Camp.php:132]: Man sollte prüfen, dass das nicht gejoined wird, wenn wir die contentnodes gar nicht brauchen

Comment (pmattmann) [reply api/src/Entity/Camp.php:132]: tut es nicht

File: api/src/Doctrine/Filter/ContentNodePeriodFilter.php [hunk 3]
```diff
@@ -16,6 +17,8 @@
 use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
 
 final class ContentNodePeriodFilter extends AbstractFilter {
+    use FiltersByCampCollaboration;
```
Comment (usu) [inline api/src/Doctrine/Filter/ContentNodePeriodFilter.php:21]: We only use this to have access to `queryBuilderAddParameters`.
Should this function go into a separate class (e.g. `QueryBuilderHelper`)?

Comment (pmattmann) [reply api/src/Doctrine/Filter/ContentNodePeriodFilter.php:21]: [c0ce12d](https://github.com/ecamp/ecamp3/pull/4935/commits/c0ce12d19527f431267464ce88957ae6f770e940)

**Review by BacLuc — APPROVED** [review]

**Review by usu — APPROVED** [review]

**Pull request merged**